gms/gossiper.cc

```diff
@@ -409,6 +409,9 @@ future<> gossiper::handle_ack2_msg(gossip_digest_ack2 msg) {
 }
 
 future<> gossiper::handle_echo_msg() {
+    if (_initial_generation_number) {
```

Comment:
This is a bit confusing since it holds an optional number so from reading it, without being aware it's optional, it looks like this expression is testing the value, rather the optional state.

I suggest calling `_initial_generation_number.has_value()` instead to make it clear.

Reply:
It is a pretty standard way to use if (foo_opt) to check if the opt contains a value. I do not mind using has_value to make it clear.

Reply:
True.  The reason it is more confusing here, is that optional value is numeric and so the member name suggests.
Until I realized it holds an optional<uint32_t>, I read it as testing the generation number itself, not its existence.